lib/edge_impulse/CMakeLists.txt

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+cmake_minimum_required(VERSION 3.13)
+
+set(EDGE_IMPULSE_DIR        ${CMAKE_BINARY_DIR}/edge_impulse)
+set(EDGE_IMPULSE_SOURCE_DIR ${EDGE_IMPULSE_DIR}/src/edge_impulse_project)
+set(EDGE_IMPULSE_BINARY_DIR ${EDGE_IMPULSE_DIR}/src/edge_impulse_project-build)
+set(EDGE_IMPULSE_STAMP_DIR  ${EDGE_IMPULSE_DIR}/src/edge_impulse_project-stamp)
+set(EDGE_IMPULSE_LIBRARY    ${EDGE_IMPULSE_BINARY_DIR}/libedge_impulse.a)
+
+file(GENERATE OUTPUT ${EDGE_IMPULSE_DIR}/compile_options.$<COMPILE_LANGUAGE>.cmake CONTENT
+"set(EI_$<COMPILE_LANGUAGE>_COMPILE_OPTIONS \"$<TARGET_PROPERTY:zephyr_interface,INTERFACE_COMPILE_OPTIONS>\")"
+)
+
+if(CONFIG_EDGE_IMPULSE_URI STREQUAL "")
+  message(FATAL_ERROR "CONFIG_EDGE_IMPULSE_URI must be specified")
+endif()
+
+set(EI_URI_PATH ${CONFIG_EDGE_IMPULSE_URI})
+
+if(NOT ${EI_URI_PATH} MATCHES "^[a-z]+://")
+  if(NOT IS_ABSOLUTE ${EI_URI_PATH})
+    set(EI_URI_PATH ${APPLICATION_SOURCE_DIR}/${EI_URI_PATH})
+  endif()
+endif()
+  set(EI_URI_PREFIX "file")
+  set(EI_URI_PATH ${APPLICATION_SOURCE_DIR}/${CONFIG_EDGE_IMPULSE_URI})
+  set(EI_URI ${EI_URI_PREFIX}://${EI_URI_PATH})
+  if(NOT EXISTS ${EI_URI_PATH})
+    message(FATAL_ERROR "Edge Impulse zip file does not exist (${EI_URI_PATH})" )
+  endif()
+else()
+  set(EI_URI ${CONFIG_EDGE_IMPULSE_URI})
+endif()
```

Comment:
This must be cleaned up, will not work.
And please keep out relative paths for time being.

Reply:
I will add the fix and squash the changes.